src/Haskell/Ide/Engine/Plugin/ApplyRefact.hs

```diff
@@ -106,6 +122,103 @@ applyAllCmd uri = pluginGetFile "applyAll: " uri $ \fp -> do
 
 -- ---------------------------------------------------------------------
 
+newtype HlintDataDir = HlintDataDir
+  { getHlintDataDir :: Maybe FilePath
+  }
+  deriving (Show, Eq, Read, Ord)
+
+
+instance ExtensionClass HlintDataDir where
+  initialValue = HlintDataDir Nothing
+
+data InitialisationError
+  = NoDataFiles FilePath
+  | InvalidDataDir FilePath
+  | Other
+
+ppInitialisationError :: InitialisationError -> String
+ppInitialisationError (NoDataFiles fp) =
+  "We could not find the data-files at: \"" <> fp <> "\""
+ppInitialisationError (InvalidDataDir fp) =
+  "The location \""
+    <> fp
+    <> "\" does not exist. We obtained that location from the environment variable: "
+    <> hlintDataDirEnvironmentVariable
+ppInitialisationError Other = "We could not determine why it failed."
+
+data Tool = Stack | Cabal
+
+showTool :: Tool -> String
+showTool Stack = "stack"
+showTool Cabal = "cabal"
+
+hlintName :: String
+hlintName = "hlint-" ++ VERSION_hlint
+
+hlintDataDirDefaultLocation :: MonadIO m => m FilePath
+hlintDataDirDefaultLocation = liftIO $ getXdgDirectory XdgData ("hie" </> "hlint")
```

Comment:
I feel like we have to duplicate this code in `install.hs` and especially to `VERSION_hlint` we have no access. Any ideas @jneira?

Reply:
we are using the same version of hlint for all ghc versions, right? It is hardcoded in all `stack-*.yaml`, so maybe it is not a big deal hardcoding it in the script? Not very elegant, though.

Mmm maybe overriding the environment variable `hlint_datadir` defined by cabal in `Paths_hlint.hs` before installing hie, so hlint data files will go in that location (not sure it works that way)